### ISSUE TITLE
Uncaught TypeError: Patch does not apply

### ISSUE THREAD

**Issue by dnapier** [t9md/atom-vim-mode-plus#1007]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.23.1 x64
**Electron**: 1.6.15
**OS**: "Arch Linux"
**Thrown From**: [vim-mode-plus](https://github.com/t9md/atom-vim-mode-plus) package 1.24.4

### Stack Trace

Uncaught TypeError: Patch does not apply

```
At /usr/lib/atom/node_modules/text-buffer/lib/default-history-provider.js:145

TypeError: Patch does not apply
    at DefaultHistoryProvider.module.exports.DefaultHistoryProvider.getChangesSinceCheckpoint (/usr/lib/atom/node_modules/text-buffer/lib/default-history-provider.js:145:22)
    at TextBuffer.module.exports.TextBuffer.getChangesSinceCheckpoint (/usr/lib/atom/node_modules/text-buffer/lib/text-buffer.js:999:42)
    at InsertAfterEndOfLine.getChangeSinceCheckpoint (/packages/vim-mode-plus/lib/operator-insert.js:25:31)
    at /packages/vim-mode-plus/lib/operator-insert.js:100:31
    at Function.module.exports.Emitter.simpleDispatch (/usr/lib/atom/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/usr/lib/atom/node_modules/event-kit/lib/emitter.js:141:28)
    at VimState.activate (/packages/vim-mode-plus/lib/vim-state.js:450:20)
    at /packages/vim-mode-plus/lib/main.js:134:39)
    at HTMLElement.didDispatch (/packages/vim-mode-plus/lib/main.js:181:30)
    at CommandRegistry.handleCommandEvent (/usr/lib/atom/src/command-registry.js:388:30)
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/lib/atom/node_modules/atom-keymap/lib/keymap-manager.js:617:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/lib/atom/node_modules/atom-keymap/lib/keymap-manager.js:408:22)
    at WindowEventHandler.handleDocumentKeyEvent (/usr/lib/atom/src/window-event-handler.js:108:34)
```

### Commands

```
     -0:41.9.0 markdown-writer:insert-new-line (input.hidden-input)
     -0:41.9.0 editor:newline (input.hidden-input)
     -0:39.4.0 core:move-down (input.hidden-input)
     -0:37.7.0 markdown-writer:insert-new-line (input.hidden-input)
     -0:37.7.0 editor:newline (input.hidden-input)
     -0:37.3.0 markdown-writer:insert-new-line (input.hidden-input)
     -0:37.3.0 editor:newline (input.hidden-input)
     -0:28.3.0 markdown-writer:insert-new-line (input.hidden-input)
     -0:28.3.0 editor:newline (input.hidden-input)
     -0:12.8.0 markdown-writer:insert-new-line (input.hidden-input)
     -0:12.8.0 editor:newline (input.hidden-input)
  2x -0:11.9.0 core:backspace (input.hidden-input)
     -0:11 core:move-down (input.hidden-input)
     -0:09.9.0 vim-mode-plus:activate-normal-mode (input.hidden-input)
     -0:09.2.0 intentions:highlight (input.hidden-input)
     -0:08.9.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.30.9 
atom-material-ui 2.0.6 
atom-react 0.4.0 
autocomplete-python 1.10.5 
build 0.70.0 
busy 0.7.0 
busy-signal 1.4.3 
color-picker 2.2.5 
css-comb 2.5.0 
git-plus 7.10.0 
hacker-dream-syntax 1.4.8 
intentions 1.1.5 
language-archlinux 0.9.2 
language-awk 0.1.12 
language-powershell 4.0.0 
language-tmux 0.4.0 
language-unix-shell 0.40.1 
linter 2.2.0 
linter-pycodestyle 2.1.3 
linter-ui-default 1.6.10 
markdown-writer 2.7.3 
minimap 4.29.7 
nord-atom-ui 0.11.0 
python-tools 0.6.9 
react 0.17.0 
script 3.17.3 
terminal-fusion 2.4.4 
vim-mode-plus 1.24.4 
wordcount 2.10.4 
```

**Comment by Ben3eeE** [comment]
This is  likely caused by a bug in `atom/text-buffer` related to `undo: skip`. The fix is currently scheduled to be released in Atom 1.25 /cc: https://github.com/atom/text-buffer/pull/282

**Comment by t9md** [comment]
Thanks @Ben3eeE for the info!
